src/components/vm/disks/diskAdd.jsx

```diff
@@ -513,11 +515,40 @@ export class AddDiskModalBody extends React.Component {
             break;
         }
         case 'file': {
+            this.setState({ file: value });
+
             if (value.endsWith(".iso")) {
                 // use onValueChange instead of setState in order to perform subsequent state change logic
                 this.onValueChanged("device", "cdrom");
             }
-            this.setState({ file: value });
+
+            if (value) {
+                cockpit.spawn(["head", "--bytes=16", value], { binary: true, err: "message", superuser: "try" })
+                        .then(file_header => {
+                            let format = "";
+
+                            if (file_header[0] == 81 && file_header[1] == 70 &&
+                                file_header[2] == 73 && file_header[3] == 251) {
+                                format = "qcow2";
+                                // All zeros, no backing file offset
+                                if (file_header.slice(8).every(bytes => bytes === 0)) {
+                                    this.setState({ customDiskVerified: true, format: format, dialogError: "" });
+                                } else {
+                                    this.dialogErrorSet(_("Importing an image with a backing file is unsupported"));
```

Comment:
Instead of setting the dialog error, I think you should set FormGroup's `helperTextInvalid`, similar to how it's done here: https://github.com/cockpit-project/cockpit-machines/blob/main/src/components/vm/disks/diskAdd.jsx#L147
This would result in the error being shown right underneath the Input field, where the cause of the error is at, something similar to this:
![Screenshot from 2022-11-17 01-28-09](https://user-images.githubusercontent.com/42733240/202324708-0760a240-820d-4379-8c6d-824a8ba705f2.png)

Instead of showing the error somewhere else in the dialog:
![Screenshot from 2022-11-17 01-25-33](https://user-images.githubusercontent.com/42733240/202324769-af3bde1f-64d8-425f-993b-b75e489d6819.png)